### PR TITLE
Fixing issue with shortcodes in HTML attributes

### PR DESCRIPTION
**Pull request by TurqDevDesign**
I've commented the lines I've added if you're interested in how the fix works.

### CODE REVIEW

**Comment by adamtootle** [comment]
@TurqDevDesign I'm getting the the error `htmlentities() expects parameter 1 to be string, array given` from [this line](https://github.com/adamtootle/staff-directory/pull/15/commits/d15e26d4a126780259fccf765c3a8b3316272885#diff-432a8e0f31a6df8272f81b1f2018abf9R242). If I change that line to `$replacers[] = htmlentities($shortcode[0]);` it seems to work as expected. Are you not seeing that on your end?

**Comment by TurqDevDesign** [comment]
@adamtootle That's odd, I am getting the same error, didn't realize because it's working perfectly for me.

**Comment by TurqDevDesign** [comment]
@adamtootle Ah, I see, the reason the error didn't show any visible results is because there weren't any quote marks in the shortcodes. The only reason I ran `htmlentities()` on the shortcodes is to prevent running the replacement procedure on quotations within the shortcode, that way we are able to replace all quotes with a string surrounded by < & >, tricking wordpress. Then we simply run `html_entity_decode()` and we can properly `do_shortcode()` without any interference, switching all the quotes back after-the-fact. 

TLDR; 
Honest mistake, I went ahead and pushed the fix.

**Comment by adamtootle** [comment]
@TurqDevDesign awesome. I hate that it had to be hacky like that, but it totally makes sense what you're doing with replacing quotes temporarily. Thanks for the fix!

**Comment by TurqDevDesign** [comment]
@adamtootle I know what you mean, but honestly 20 lines of code (when you exclude my comments and white space) is one of the more convenient fixes. It's hacky, but I'd like to believe it's elegant. Not to mention that it maintains the integrity of future, standardized shortcodes without breaking old ones. It would be nice if Wordpress would just develop a patch already, but I believe this specific issue has existed for at least 5 years. I don't hold out any hope. 

And not a problem at all, it's been my pleasure to help out. Not to mention that half the bugs we've tackled these past few days were my doing, it'd be rude to make a mess and not offer a helping-hand with the clean up xD

This also reminds me, there was one more issue opened earlier, not sure if you saw it. There was a caching issue in 1.0.3, and I believe that's why peoples' templates kept disappearing. What I think was happening was that the template was stored in the database, but because of the escaping method (only adding slashes), the code was interfering with something in `get_option()`. This meant that the code could not actually be retrieved from the database, the only reason it worked was because Wordpress keeps its own cache and seemed to be loading the option from the cached options, which must be stored differently. This explains why people's templates were being wiped when they deactivated/reactivated the plugin. The cache was gone, the settings page couldn't load anything, and they'd maybe try to save when the custom html field was empty, thinking it'd bring it back. But in reality, until they pressed that save button, the template was in the database, unretrievable, and saving is what truly overwrote it. I solved this by running `htmlentities()` on the templates before storing them. 

Now I know that all was a little long-winded, but the reason I explain it is because the issue posted earlier showed that updating from 1.0.3 could cause it too. Because it clears the option cache when you update, and without re-saving their templates under the new system, the templates are still stored improperly and can not be loaded into the settings page. So with all of that backstory, I think it's important to either add some sort of notice and a how-to on how to restore templates without accidentally saving an empty field over it, or some temporary function that will pull the database content, `html_entity_decode()`, and re-`htmlentities()` on it whenever the plugin is loaded. This would ensure that as people update they will not be losing their templates. Only issue is that we'd have to leave the function in until we're sufficiently convinced that most people whom ever plan to update are updated.
